src/modules/m_sqloper.cpp

```diff
@@ -19,46 +20,104 @@
 
 #include "inspircd.h"
 #include "modules/sql.h"
-#include "modules/hash.h"
 
-class OpMeQuery : public SQLQuery
+class OpGetQuery : public SQLQuery
 {
  public:
+	// This variable will store all the OPER blocks that are from the config file. This is so we are always aware of which OPER blocks aren't from this module.
+	std::vector<std::string>& my_blocks;
+	/* We want to store the username and password if it's because of an /OPER, as this SQL class will never return anything to the SQLOper module, meaning we're responsible for actually executing the /OPER command post-DB fetch.
+	 * Note: uid will be empty if this DB update was not called as a result of a user command (i.e. /REHASH) */
 	const std::string uid, username, password;
-	OpMeQuery(Module* me, const std::string& u, const std::string& un, const std::string& pw)
-		: SQLQuery(me), uid(u), username(un), password(pw)
+	OpGetQuery(Module* me, std::vector<std::string>& mb, const std::string& u, const std::string& un, const std::string& pw)
+		: SQLQuery(me)
+		, my_blocks(mb)
+		, uid(u)
+		, username(un)
+		, password(pw)
+	{
+	}
+	OpGetQuery(Module* me, std::vector<std::string>& mb)
+		: SQLQuery(me)
+		, my_blocks(mb)
 	{
 	}
 
 	void OnResult(SQLResult& res) CXX11_OVERRIDE
 	{
-		ServerInstance->Logs->Log(MODNAME, LOG_DEBUG, "result for %s", uid.c_str());
-		User* user = ServerInstance->FindNick(uid);
-		if (!user)
-			return;
+		// Make it so we can use a legible variable
+		ServerConfig::OperIndex* oper_blocks =& ServerInstance->Config->oper_blocks;
```

Comment:
=& is not an operator. The space should be before the & not after.

Better still, can oper_blocks just be a reference?